### PR TITLE
✅ add case of test when api of survey return error #418

### LINKED ISSUE #418: Increase coverage in AnswerSurvey service
Currently dont have a test case when the api return a error, we need test to this case to check if no have surveys in screen when on error in api.

### PR DESCRIPTION
**Pull request by iurynogueira**
fix #418

#add case of test when api of survey return error

Test on error of api to register/start survey

### CODE REVIEW

File: frontend/src/components/main/surveys-list.test.tsx [hunk 1]
```diff
@@ -69,4 +69,23 @@ describe('Surveys list', () => {
       })
     );
   });
+
+  it('not should show a survey when api return error', async () => {
```
Comment (mizek1) [inline frontend/src/components/main/surveys-list.test.tsx:73]:
```suggestion
  it('should not show a survey when api return error', async () => {
```

File: frontend/src/components/main/surveys-list.test.tsx [hunk 2]
```diff
@@ -69,4 +69,23 @@ describe('Surveys list', () => {
...
+        return Promise.reject({ response: { data: mockResponse } });
+      })
+    );
+
+    await waitFor(() =>
+      expect(httpClient.post).toHaveBeenCalledWith('api/v1/answers_surveys', {
+        survey_id: 1,
+      })
+    );
+    expect(screen.queryByText('Animals survey')).toBeNull();
```
Comment (edimossilva) [inline frontend/src/components/main/surveys-list.test.tsx:89]: Does make sense to check for some error message? Or does this go outside of unit test scope?

**Review by edimossilva — APPROVED** [review]
Niceeee!! always good to see test coverage being improved!

File: frontend/src/components/main/surveys-list.test.tsx (hunk 2 above)
Comment (edimossilva) [reply frontend/src/components/main/surveys-list.test.tsx:89]: @iurynogueira

File: frontend/src/components/main/surveys-list.test.tsx (hunk 1 above)
Comment (edimossilva) [reply frontend/src/components/main/surveys-list.test.tsx:73]: @mizek1 , you are welcome to contribute as well, please consider working on some issues ❤️

**Review by edimossilva — APPROVED** [review]
Nice!!!!!

File: frontend/src/components/main/surveys-list.test.tsx (hunk 1 above)
Comment (mizek1) [reply frontend/src/components/main/surveys-list.test.tsx]: I will @edimossilva 😄

**Pull request merged**